internal/configs/transportserver.go

```diff
@@ -164,6 +179,29 @@ func generateTransportServerHealthCheckWithDefaults(up conf_v1alpha1.Upstream) *
 		Interval: "5s",
 		Passes:   1,
 		Fails:    1,
+		Match:    "",
+	}
+}
+
+func generateHealthCheckMatch(match *conf_v1alpha1.Match, name string) *version2.Match {
+	var modifier string
+	var expect string
+
+	if strings.HasPrefix(match.Expect, "~*") {
+		modifier = "~*"
+		expect = strings.TrimPrefix(match.Expect, "~*")
+	} else if strings.HasPrefix(match.Expect, "~") {
+		modifier = "~"
+		expect = strings.TrimPrefix(match.Expect, "~")
+	} else {
+		expect = match.Expect
+	}
+
+	return &version2.Match{
+		Name:                name,
+		Send:                match.Send,
+		ExpectRegexModifier: modifier,
```

Comment:
Looks like this field has two values, `~*` and `*` and based on https://serverfault.com/questions/801356/what-is-the-difference-between-nginx-and-regexes it means it's either case insensitive or case sensitive. If that's the case, can we add a field to `Match` called caseSensitiveMatch? Then we can set the `ExpectRegexModifier` based on that.

Reply:
There are 3 cases - no regex, regex with case sensitive, regex with case insensitive , so just a boolean field will not work. Also see https://nginx.org/en/docs/stream/ngx_stream_upstream_hc_module.html#match 

The current approach is consistent with the existing configuration:

```yaml
  - path: ~ ^/decaf/.*\\.jpg$
    action:
      pass: coffee
```

```yaml
  - path: /coffee
    matches:
    - conditions:
      - cookie: user
        value: ~^m
      action:
        pass: coffee-v2
    action:
      pass: coffee-v1
```

Reply:
Can we add a new, optional, field called 'regexModifier' with default of empty? 
So it looks like:
```
match:
   expect:
   send:
   regexModifier:
```
I think we shouldn't compress values and modifiers into the same field. In terms of design, it's easier to change and easier to reason about.

Reply:
As mentioned here https://github.com/nginxinc/kubernetes-ingress/pull/1633#discussion_r640920775, the choice was made to towards consistency with the existing configuration:
```yaml
  - path: ~ ^/decaf/.*\\.jpg$ # regex
    action:
      pass: coffee
  - path: /coffee
    matches:
    - conditions:
      - cookie: user
        value: ~^m # regex
      action:
        pass: coffee-v2
    action:
      pass: coffee-v1
```

Note that YAML configuration is independent of how we represent the NGINX config in the structs we use for templates. And In those structs, the regexModifier field is present, as it makes it easier to generate the config.